### PR TITLE
Add python bindings

### PR DESCRIPTION
**Pull request by Kwaizer**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Draft PR to add python bindings by making a multi-output feedstock

### CODE REVIEW

**Comment by matthewfeickert** [comment]
We should fix the rerender in PR https://github.com/conda-forge/hepmc2-feedstock/pull/6 and then rebase this PR to keep things more atomic.

**Comment by Kwaizer** [comment]
Adding `- {{ stdlib("c") }}` as proposed in documentation results in `c_linux-64 does not exist` error for some reason

**Comment by chrisburr** [comment]
@conda-forge-admin, please rerender

**Comment by matthewfeickert** [comment]
@Kwaizer now that PR #7 is in, can you rebase this PR on `main` but do so in an interactive rebase that drops all of the `MNT: Re-rendered` commits as those have already been taken care of? This will make it easier to review what is actually getting changed in this PR.

**Comment by matthewfeickert** [comment]
@conda-forge/hepmc2 can you review this when you have time?

File: recipe/meta.yaml [hunk 1]
```diff
@@ -20,6 +20,15 @@ requirements:
     - {{ stdlib("c") }}
     - cmake
     - make
+    - python
+    - cython
```
Comment (matthewfeickert) [inline recipe/meta.yaml]: Can you comment on why `cython` is needed for the build? That isn't a formal dependency  to build normally (c.f. https://github.com/matthewfeickert/hep-simulation-stack/blob/4c91a3ea81891b05fd8f38df78f46fcc824bb71e/docker/Dockerfile#L63-L81).

File: recipe/meta.yaml [hunk 2]
```diff
@@ -20,6 +20,15 @@ requirements:
...
+  host:
+    - python
+    - cython
+    - setuptools
+    - pip
```
Comment (matthewfeickert) [inline recipe/meta.yaml]: In addition to `cython`, why are `setuptools` and `pip` needed as [`host` requirements](https://docs.conda.io/projects/conda-build/en/stable/resources/define-metadata.html#host)?

File: recipe/meta.yaml [hunk 3]
```diff
@@ -38,6 +49,8 @@ test:
     - ${CXX} example_EventSelection.cc -o example_EventSelection -I${CONDA_PREFIX}/include -L${CONDA_PREFIX}/lib -lHepMC
     # example_BuildEventFromScratch
     - ${CXX} example_BuildEventFromScratch.cc -o example_BuildEventFromScratch -I${CONDA_PREFIX}/include -L${CONDA_PREFIX}/lib -lHepMC
+    # temporarily skipping pip check on win and pypy build because of bizarre tqdm issue
```
Comment (matthewfeickert) [inline recipe/meta.yaml]: Can you post the `tqdm` issue that you're seeing?

**Review by matthewfeickert — COMMENTED** [review]
Thanks for rebasing @Kwaizer &mdash; this will be much easier to review for the maintainer team now. :+1: Going through this looks good, but I didn't follow on some of the added requirements. Can you elaborate a bit?

Can you also bump the build number from 

```yaml
build:
  number: 1
```

to `2`?

File: recipe/meta.yaml (hunk 2 above)
Comment (Kwaizer) [reply recipe/meta.yaml]: Those dependencies where installed in attempt to fix issues while trying to build multiple-output package and can be removed for now. 
I think that switching to packaging another tools or working in parallel on both `hepmc2` and `LHAPDF` (or other tool)  will be a good idea as multiple-output package takes way more time than I expected for some reason.

File: recipe/meta.yaml (hunk 3 above)
Comment (Kwaizer) [reply recipe/meta.yaml]: Oh, sorry for a confusion. That was a copy-pasted code snippet when I was figuring out how testing works for a package. It have nothing to do with this particular package, I simply forgot to remove it.

**Review by matthewfeickert — COMMENTED** [review]
So at the moment the build script is building with the defaults

https://github.com/conda-forge/hepmc2-feedstock/blob/9ac500d83544bf5e79dc04ee57696c3592273121/recipe/build.sh#L10-L14

and I don't think that includes the Python bindings. What we'll want to do is to try to create a multiple output feedstock like this one @chrisburr shared (https://github.com/conda-forge/brotli-feedstock/blob/1971636849d467a838e84f4a19b027d445e9e59a/recipe/meta.yaml) and then as the `hepmc2-python` output build with the Python bindings enabled.

I only remember how to do this for HepMC3, so you'll need to look at the HepMC2 GitLab to see how they enable Python bindings there.

**Comment by matthewfeickert** [comment]
Given this comment from Andy Buckely (https://github.com/research-software-collaborations/project_database/pull/62#discussion_r1713599070) we can close this PR.

**Comment by matthewfeickert** [comment]
@conda-forge/hepmc2 can you close this PR, as this won't be supported?

**Pull request closed without merging**